From 6b1801dbd2fd0997e032f365e549dffd4be432e7 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 29 Nov 2021 12:38:46 +0200
Subject: [PATCH] [SW-63789] gaudi2: update sp reg19 based on intr type

In order for validation to identify RAZWI, HBW and
LBW pcie rr interrupt, a unique pattern is written
to scratchpad register 19.
RAZWI- pattern 0xABBA
HBW - pattern 0xDEAF
LBW - pattern 0xACDC

NOTE: This is a revert me patch and the logic will be removed
post Gaudi2 PO validation.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: Idf69440f9bcde461f8194a1059f23a3597fb0054
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index a52de69..85ff6c3 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1030,11 +1030,19 @@
  */
 void check_razwi_intr(void)
 {
+	u32 pattern = 0;
 	u32 reg = regmap_readl(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
 	u32 bitmask = reg & PCIE_INTR_ERR_INTR_BIT;
 
-	if (bitmask) {
-		regmap_writel(RAZWI_INTR_STATUS_PATTERN, mmGIC_RAZWI_STATUS_REG);
+	if (bitmask & PCIE_HBW_ERR_INTR_BIT)
+		pattern = PCIE_HBW_INTR_STS_PATTERN;
+	else if (bitmask & PCIE_LBW_ERR_INTR_BIT)
+		pattern = PCIE_LBW_INTR_STS_PATTERN;
+	else if (bitmask & PCIE_BAD_ACC_ERR_INTR_BIT)
+		pattern = RAZWI_INTR_STATUS_PATTERN;
+
+	if (pattern) {
+		regmap_writel(pattern, mmGIC_RAZWI_STATUS_REG);
 		/* Reset the indication */
 		regmap_writel(reg & ~bitmask, mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
 	}
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index d3e1b66..3795e74 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -63,7 +63,13 @@
  * and to be removed after GAUDI2 PO
  */
 #define PCIE_INTR_ERR_INTR_BIT		(BIT(0) | BIT(1) | BIT(2))
+#define PCIE_HBW_ERR_INTR_BIT		BIT(0)
+#define PCIE_LBW_ERR_INTR_BIT		BIT(1)
+#define PCIE_BAD_ACC_ERR_INTR_BIT	BIT(2)
 #define RAZWI_INTR_STATUS_PATTERN	0xABBA
+#define PCIE_HBW_INTR_STS_PATTERN	0xDEAF
+#define PCIE_LBW_INTR_STS_PATTERN	0xACDC
+
 #define NUM_OF_MME_WB_PORTS		2
 
 enum nic_mac_lane {
